src/components/Reown/modals/SendNanoContractTxModal.js

```diff
@@ -0,0 +1,255 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { t } from 'ttag';
+import { useDispatch, useSelector } from 'react-redux';
+import { types, setNewNanoContractStatusReady } from '../../../actions';
+import helpers from '../../../utils/helpers';
+import nanoUtils from '../../../utils/nanoContracts';
+import { NanoContractActions } from '../NanoContractActions';
+import { getGlobalWallet } from '../../../modules/wallet';
+import AddressList from '../../AddressList';
+import { NANO_UPDATE_ADDRESS_LIST_COUNT } from '../../../constants';
+
+export function SendNanoContractTxModal({ data, firstAddress, onAccept, onReject }) {
+  const dispatch = useDispatch();
+  const blueprintInfo = useSelector((state) => state.blueprintsData[data?.data?.blueprintId]);
+  const nanoContracts = useSelector((state) => state.nanoContracts);
+  const decimalPlaces = useSelector((state) => state.serverInfo.decimalPlaces);
+  const [selectedAddress, setSelectedAddress] = useState(firstAddress);
+  const [isSelectingAddress, setIsSelectingAddress] = useState(false);
+
+  // Ensure we have a valid address by loading the first one if needed
+  useEffect(() => {
+    const ensureValidAddress = async () => {
+      if (!selectedAddress) {
+        const wallet = getGlobalWallet();
+        if (wallet.isReady()) {
+          const address = await wallet.getAddressAtIndex(0);
+          setSelectedAddress(address);
+        }
+      }
+    };
```

Comment:
Why do you need this is `firstAddress` is one of the props of this component?

Reply:
We don't need this, and we even had a bug, causing the caller to always be the `firstAddress`

Can you check 484ec080fefaf799e4e85162fc75494a3f19f824?